charts/pwa/templates/deployment.yaml

```diff
@@ -11,6 +11,8 @@ metadata:
     {{- toYaml .Values.podLabels | nindent 4 }}
     {{- end }}
 spec:
+  strategy:
+    type: Recreate
```

Comment:
Has this been requested for a reason? Because this will terminate all old pod revisions prior to deploying the new revisions in the system.

Reply:
Yes. In my time working for an Intershop partner we encountered problems when the newly started nginx was caching content from the still running but old PWA containers, caused by a rolling update.

Recreate will cause downtime, but usually our deployments were scheduled, as they also included ICM database migrations.

Reply:
True. In that case I think this should be mentioned as a known limitation. Something like updates without downtime are not possible at the moment.

Reply:
This should definitely be mentioned, since most of our customers use rolling update as a strategy to deploy only hotfixes, without downtime. In my opinion, it should be implemented in such a way, that the current behavior does not change, but instead to make the option for recreating the Pods optional for people who want to use it.

Otherwise, it should be mentioned with **BIG BOLD LETTERS** that upgrades without downtime will not be possible with this HelmChart Version.

Reply:
May you can make the update-strategy configurable. This would provide the capability to use rolling updates in case there is no cache reset necessary.

Reply:
> May you can make the update-strategy configurable. This would provide the capability to use rolling updates in case there is no cache reset necessary.

I think it would be best to have the current behavior as default and be able to configure the new behavior as optional. Since most consumers of our HelmChart don't have the update strategy in their HelmReleases and instead count on the default update strategy (which is currently rolling update). 
That way we won't have to reconfigure like 150 HelmReleases but only the few ones who want to have the recreate strategy should actively change their helm values to enable it.

Reply:
Please do it somehow unique to https://github.com/intershop/helm-charts/blob/87b9be70b96d6b02395d1bb29aed37265e936f53/charts/icm-as/values.yaml#L7C1-L8C30 to not have too much differences between each helm chart.

Reply:
Thank you for your comments, I updat4ed the PR to have a configurable update strategy defaulting to rolling. This way no breaking changes are introduced.